### ISSUE TITLE
Authentication still fails when using a credential_process and source_profile

### ISSUE THREAD

**Issue by michael-plutoflume** [aws/aws-toolkit-vscode#1655]
**Describe the bug**

I get an error when trying to use a credential_process in conjunction with a source_profile.

**To Reproduce**

I have a config file (~/.aws/config) that is equivalent to

```
[profile base-profile]
credential_process = some-process

[profile child]
region            = eu-west-1
role_arn          = arn:aws:iam::123456789:role/some-role
source_profile    = base-profile
role_session_name = something
```

and no credentials file. This works fine with the CLI, python SDK etc. but when I try and select the child profile for use with the toolkit I get an error. The logs read

```
2021-04-16 09:22:04 [ERROR]: Error trying to connect to AWS with Credentials Provider profile:child. Toolkit will now disconnect from AWS. [Error [CredentialsError]: Missing credentials in config, if using AWS_CONFIG_FILE, set AWS_SDK_LOAD_CONFIG=1
	at constructor.load (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1649:6028)
	at constructor.coalesceRefresh (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1438:26701)
	at constructor.refresh (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1649:6223)
	at constructor.get (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1438:26529)
	at f (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1638:27945)
	at constructor.getCredentials (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1638:28226)
	at constructor.<anonymous> (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1638:30721)
	at constructor.callListeners (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1438:24989)
	at constructor.emit (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1438:24739)
	at constructor.emitEvent (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1639:10845)
	at constructor.N (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1639:6431)
	at u.runTo (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1639:12683)
	at constructor.runTo (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1639:8122)
	at constructor.send (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1639:8015)
	at r.constructor.r.constructor.makeRequest (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1638:17058)
	at r.constructor.r.constructor.e.authtype.d.<computed> [as assumeRole] (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1638:23928)
	at constructor.loadRoleProfile (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1649:7497)
	at constructor.load (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1649:5604)
	at constructor.coalesceRefresh (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1438:26701)
	at constructor.refresh (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1649:6223)
	at constructor.get (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1438:26529)
	at new constructor (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1649:5199)
	at /home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:2204:3416
	at f (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1438:27709)
	at constructor.resolve (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:1438:27806)
	at /home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:364:15558
	at new Promise (<anonymous>)
	at constructor.resolvePromise (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:364:15498)
	at Sd.getCredentials (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:2204:1797)
	at Wk.setCredentials (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:2622:587)
	at Wk.upsertCredentials (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:2622:449)
	at async Mv.login (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:2216:327)
	at async K_.onCommandLogin (/home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:2526:310)
	at async /home/michael.aldridge/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.23.0/dist/extension.js:2934:1140
	at async S._executeContributedCommand (/usr/share/code/resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:86:106762)] {
  code: 'CredentialsError',
  time: 2021-04-16T08:22:04.774Z,
  originalError: {
    message: 'Could not load credentials from constructor',
    code: 'CredentialsError',
    time: 2021-04-16T08:22:04.774Z,
    originalError: {
      message: 'Credentials not set for profile base-profile',
      code: 'SharedIniFileCredentialsProviderFailure',
      time: 2021-04-16T08:22:04.774Z
    }
  }
}
```

**Expected behavior**

Authentication to succeed.

**Desktop (please complete the following information):**

- OS: Linux x64 5.4.0-70-generic
- Visual Studio Code Extension Host Version: 1.55.0
- AWS Toolkit Version: 1.23.0

**Additional context**

This issue is very similar to https://github.com/aws/aws-toolkit-vscode/issues/1541 that was addressed in 1.23.

**Comment by JadenSimon** [comment]
Hey @michael-plutoflume,

I was able to replicate this bug, and it appears to be a limitation of the AWS JS SDK. The SDK is unable to load 'credential_process' when using the SDK's `SharedIniFileCredentials`. Currently, the only way to use 'source_profile' or 'credential_source' is by using `SharedIniFileCredentials`. This issue can be tracked here: https://github.com/aws/aws-sdk-js/issues/3247

Let us know if you need anymore help!

**Comment by michael-plutoflume** [comment]
Understood. Thanks for looking into this @JadenSimon.

**Comment by DanyC97** [comment]
i think i'm hitting similar issue although the logs i get back are a bit different

```
021-05-08 17:37:21 [ERROR]: Error trying to connect to AWS with Credentials Provider profile:stage-acc1. Toolkit will now disconnect from AWS. [ProcessCredentialsProviderFailure: credential_process returned error
	at /Users/user1/.vscode/extensions/amazonwebservices.aws-toolkit-vscode-1.24.0/dist/extension.js:1441:5196
	at ChildProcess.exithandler (child_process.js:324:5)
	at ChildProcess.emit (events.js:315:20)
	at ChildProcess.EventEmitter.emit (domain.js:467:12)
	at maybeClose (internal/child_process.js:1048:16)
	at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)] {
  code: 'ProcessCredentialsProviderFailure',
  time: 2021-05-08T16:37:21.040Z
}

```

My `~/.aws/credentials` looks the same as @michael-plutoflume.

@JadenSimon is the toolkit using sdk-js-v2 or -v3 ?  Anything you can do internally to bump the priority of a) either getting https://github.com/aws/aws-sdk-js/issues/3247 fixed or b) migrate to v3 and see if that works better?

Otherwise @JadenSimon  is it possible to add a note in the docs https://docs.aws.amazon.com/toolkit-for-vscode/latest/userguide/external-credential-process.html and say "known issue - it doesn't work because of x,y,z". I've spent good amount of time trying to understand why is not working although the docs were saying _it should_

**Comment by justinmk3** [comment]
> is the toolkit using sdk-js-v2 or -v3 ?

Toolkit currently uses v2. We have started migrating to v3 but it will take some time.
 
> is it possible to add a note in the docs https://docs.aws.amazon.com/toolkit-for-vscode/latest/userguide/external-credential-process.html and say "known issue - it doesn't work because of x,y,z"

Good idea, will do.

**Comment by JadenSimon** [comment]
@DanyC97 

I was able to verify that this issue is fixed in v3 of the SDK. Adding a 'known issues' section is a great idea until we can finish migration, thank you for the suggestion!

**Comment by JadenSimon** [comment]
@michael-plutoflume @DanyC97 
We've been working on a partial migration to SDK v3 that should fix various combinations of `source_profile` and profile types. 

To confirm the fix works as intended, it would be helpful if one of you can try out the pre-release build:

1. Download and extract [buildArtifacts.zip](https://github.com/aws/aws-toolkit-vscode/files/6540621/buildArtifacts.zip)
2. From the VSCode command list (ctrl-shift-p or cmd-shift-p) run Extensions: Install from VSIX... to install the *.vsix file.
3. Reload VSCode and verify that the Toolkit has version 1.26 (not the release version 1.25).

**Comment by michael-plutoflume** [comment]
> @michael-plutoflume @DanyC97
> We've been working on a partial migration to SDK v3 that should fix various combinations of `source_profile` and profile types.
> 
> To confirm the fix works as intended, it would be helpful if one of you can try out the pre-release build:
> 
> 1. Download and extract [buildArtifacts.zip](https://github.com/aws/aws-toolkit-vscode/files/6540621/buildArtifacts.zip)
> 2. From the VSCode command list (ctrl-shift-p or cmd-shift-p) run Extensions: Install from VSIX... to install the *.vsix file.
> 3. Reload VSCode and verify that the Toolkit has version 1.26 (not the release version 1.25).

Hi @JadenSimon just gave this a test and it seems to be working. :tada:

**Comment by v-au** [comment]
Works here as well.

**Comment by justinmk3** [comment]
Fixed in AWS Toolkit 1.29
